scene/main/http_request.cpp

```diff
@@ -159,16 +159,21 @@ void HTTPRequest::cancel_request() {
 	if (!use_threads) {
 		set_process_internal(false);
 	} else {
-		thread_request_quit = true;
-		Thread::wait_to_finish(thread);
-		memdelete(thread);
-		thread = NULL;
+
+		if (thread) {
```

Comment:
I think a better solution would be to fix the condition in `set_use_threads` that can cause the value to change before the request has finished, thus causing `use_threads` to be `true` but `thread` to be null. (`_request_done`/`cancel_request` is called deferred when status become disconnected during poll, thus causing a frame to be in the situation where `status == STATUS_DISCONNECTED` but `requesting == true`)

```diff
diff --git a/scene/main/http_request.cpp b/scene/main/http_request.cpp
index e65314644..d6392d62d 100644
--- a/scene/main/http_request.cpp
+++ b/scene/main/http_request.cpp
@@ -425,7 +425,7 @@ void HTTPRequest::_notification(int p_what) {
 
 void HTTPRequest::set_use_threads(bool p_use) {
 
-       ERR_FAIL_COND(status != HTTPClient::STATUS_DISCONNECTED);
+       ERR_FAIL_COND(requesting);
        use_threads = p_use;
 }
```

Reply:
I think, thats a bad idea to remove that `STATUS_DISCONNECTED` because someone can be easily confused at that line. Intead, should we check the `requesting` status that way? @akien-mga 

```
if (!requesting) {
	use_threads = p_use;
}
```

Reply:
> Isn't this exactly the change I proposed above in the diff?

Oh, I see the difference, keeping both COND, sorry. Yeah, we might keep both conditions. I think it needs to print an error though if you try to change it while requesting.

Reply:
I think I found the underlying issue. See #30218

Reply:
The changes in this file can now be removed since #30218 is merged.

Reply:
Looks like @Faless 's way is more effective solution. I removed that changes.